Tools/cases_generator/generate_cases.py

```diff
@@ -549,10 +549,23 @@ def write_metadata(self, metadata_filename: str, pymetadata_filename: str) -> No
                 "=",
                 ";",
             ):
-                for name, _ in self.families.items():
+                family_member_names: set[str] = set()
```

Comment:
Do you still need to iterate over families if you're also iterating over the individual instructions?

Reply:
At the very least I need to collect the lists of family (non-head) members, so I can tell which macros need cache metadata. I'll look into whether I can emit all metadata in the following loops though. (This is perhaps an area where Mark's desire to translate instr -> op+macro in an early pass would simplify the code.)